### ISSUE TITLE
Allow for a different app_build_command to be run between staging and production envs

### ISSUE THREAD

**Issue by cfu288** [Azure/static-web-apps#218]
Loving static web apps so far!

Right now we have the ability to set different env vars between staging and prod azure functions (unfortunately still a manual process) but no way to define custom envs for the front end app. This could be solved by allowing users so specify a custom app_build_command in their Github Actions workflow file that would be run when deploying to stage. 

For example, a user could specify that their CRA React app should run `npm build` for prod but `npm build:staging` for staging or something along those lines.

**Comment by miwebst** [comment]
Hi @cfu288 this is great suggestion! I assume you are looking for this improvement with respect to the PRs producing an environment feature? Today you can specify an app_build_command but that will apply to all environments of a particular static web app. You could also accomplish what you are trying to do with 2 static web apps with one specifying an app build command for staging.

**Comment by cfu288** [comment]
Yup, I meant that it would be great if PR's could create a new staging environment by running a different build command than the one that is run for prod. I did think about creating two static web apps, but that seemed clunkier and requires a branch per static web app, making trunk based development more difficult. 

This is by no means a blocker for me, just more of a feature request.

**Comment by miwebst** [comment]
Makes sense! Also just as an fyi, you can have multiple static web apps attached to the same repo/branch so you could keep a single branch for multiple apps if you'd like!

**Comment by cfu288** [comment]
Oh is that the case? I was having trouble putting multiple apps on the same branch. On deploy I would get an error:

`
A static site already exists for the given repository and branch. The existing static site is /<redacted>/Microsoft.Web/staticSites/spaceinterns-webapp
`

I was able to create multiple apps from the same repo, just not the same branch. Let me know if I'm missing something!

**Comment by miwebst** [comment]
Ah this is due to a bug on our side; the fix is still rolling out unfortunately. If you delete the existing static web app on that repo/branch, then you should be able to create multiple static web apps against that repo/branch going forward.
